packages/fuzzer/fuzzer.ts

```diff
@@ -14,191 +14,27 @@
-	let shouldCallLibfuzzer = false;
-	switch (operator) {
-		case "==":
-			result = s1 == s2;
-			shouldCallLibfuzzer = !result;
-			break;
-		case "===":
-			result = s1 === s2;
-			shouldCallLibfuzzer = !result;
-			break;
-		case "!=":
-			result = s1 != s2;
-			shouldCallLibfuzzer = result;
-			break;
-		case "!==":
-			result = s1 !== s2;
-			shouldCallLibfuzzer = result;
-			break;
-	}
-	if (shouldCallLibfuzzer && s1 && s2) {
-		addon.traceUnequalStrings(id, s1, s2);
-	}
-	return result;
-}
-
-/**
- * Performs an integer comparison between two strings and calls the corresponding native hook if needed.
- * This function replaces the original comparison expression and preserves the semantics by returning
- * the original result after calling the native hook.
- * @param n1 first compared number
- * @param n2 second compared number
- * @param operator the operator used in the comparison
- * @param id an unique identifier to distinguish between the different comparisons
- * @returns result of the comparison
- */
-function traceNumberCmp(
-	n1: number,
-	n2: number,
-	operator: string,
-	id: number
-): boolean {
-	if (Number.isInteger(n1) && Number.isInteger(n2)) {
-		addon.traceIntegerCompare(id, n1, n2);
-	}
-	switch (operator) {
-		case "==":
-			return n1 == n2;
-		case "===":
-			return n1 === n2;
-		case "!=":
-			return n1 != n2;
-		case "!==":
-			return n1 !== n2;
-		case ">":
-			return n1 > n2;
-		case ">=":
-			return n1 >= n2;
-		case "<":
-			return n1 < n2;
-		case "<=":
-			return n1 <= n2;
-		default:
-			throw `unexpected number comparison operator ${operator}`;
-	}
-}
-
-function traceAndReturn(current: unknown, target: unknown, id: number) {
-	switch (typeof target) {
-		case "number":
-			if (typeof current === "number") {
-				if (Number.isInteger(current) && Number.isInteger(target)) {
-					addon.traceIntegerCompare(id, current, target);
-				}
-			}
-			break;
-		case "string":
-			if (typeof current === "string") {
-				addon.traceUnequalStrings(id, current, target);
-			}
-	}
-	return target;
-}
-
-// Re-export everything from the native library.
-export type FuzzTargetAsyncOrValue = (data: Buffer) => void | Promise<void>;
-export type FuzzTargetCallback = (
-	data: Buffer,
-	done: (e?: Error) => void
-) => void;
-export type FuzzTarget = FuzzTargetAsyncOrValue | FuzzTargetCallback;
-export type FuzzOpts = string[];
+export type {
+	FuzzTarget,
+	FuzzTargetAsyncOrValue,
+	FuzzTargetCallback,
+} from "./addon";
 
 export interface Fuzzer {
-	printVersion: () => void;
-	startFuzzing: (fuzzFn: FuzzTarget, fuzzOpts: FuzzOpts) => void;
-	startFuzzingAsync: (fuzzFn: FuzzTarget, fuzzOpts: FuzzOpts) => Promise<void>;
-	stopFuzzingAsync: (status?: number) => void;
-	nextCounter: typeof nextCounter;
-	incrementCounter: typeof incrementCounter;
-	readCounter: typeof readCounter;
-	traceStrCmp: typeof traceStrCmp;
-	traceNumberCmp: typeof traceNumberCmp;
-	traceAndReturn: typeof traceAndReturn;
+	nativeAddon: NativeAddon;
```

Comment:
I really like using properties to encapsulate related functions in `Fuzzer`.

But, IMHO, `NativeAddon` is an implementation detail and shouldn't be exposed outside of the `fuzzer` package. Its function could be divided between `coverageTracer`, `tracer` and another property related to the fuzzer execution / state. WDYT?